### ISSUE TITLE
What's the best way for a total n00b to start with this library?

### ISSUE THREAD

**Issue by jgeewax** [GoogleCloudPlatform/gcloud-java#34]
I'm having trouble getting a HelloDatastore project working. Hoping that you can help me get this going...

Here's what I did:

```bash
$ git clone https://github.com/GoogleCloudPlatform/gcloud-java.git
$ cd gcloud-java
$ sudo apt-get install maven
$ mvn install
$ export CLASSPATH=~/gcdjavasample/gcloud-java/target/gcloud-java-0.0.3.jar
```

Here's the file I'm trying to run (HelloDatastore.java):

```java
import com.google.gcloud.datastore.DatastoreService;
import com.google.gcloud.datastore.DatastoreServiceFactory;
import com.google.gcloud.datastore.DatastoreServiceOptions;
import com.google.gcloud.datastore.Entity;
import com.google.gcloud.datastore.Key;
import com.google.gcloud.datastore.KeyFactory;



public class HelloDatastore {
  private static final String DATASET = "gcloud-datastore-demo";

  public static void main(String[] args) {
    DatastoreServiceOptions options = DatastoreServiceOptions.builder().dataset(DATASET).build();
    DatastoreService datastore = DatastoreServiceFactory.getDefault(options);
    KeyFactory keyFactory = datastore.newKeyFactory().kind("Person");
    Key key = keyFactory.newKey("Jimmy");

    System.out.println("Trying to get the entity by its key!");

    Entity entity = datastore.get(key);

    if (entity == null) {
      System.out.println("Entity not found! Creating it!");
      entity = Entity.builder(key)
          .set("name", "John Doe")
          .set("age", 30)
          .set("updated", false)
          .build();
      datastore.put(entity);
    } else {
      System.out.println("Entity found! Updating it!");
      boolean updated = entity.getBoolean("updated");
      if (!updated) {
        String[] name = entity.getString("name").split(" ");
        entity = Entity.builder(entity)
            .set("name", name[0])
            .set("last_name", name[1])
            .set("updated", true)
            .remove("old_property")
            .set("new_property", 1.1)
            .build();
        datastore.update(entity);
      }
    }

    System.out.println("Done!");
  }
}
```

I can get it to compile just fine, then I run into problems running the class. Apparently I need Guava? And google-auth-library-java ? And google-api-java-client ?

What exactly is the best way to get this so that I can run... `java HelloDatastore` and I'll see some stuff happening.... ? 

(Sorry for the dumb question....)

**Comment by aozarov** [comment]
When using maven one's artifact does not include/bundle all its dependencies.
There are various ways to deal with it but the most recommended way for someone that consumes
a maven artifact[s] is to make its code using maven (http://maven.apache.org/guides/getting-started/maven-in-five-minutes.html) and adding gcloud-java as a dependency as such:
<dependency>
  <groupId>com.google.gcloud</groupId>
  <artifactId>gcloud-java</artifactId>
  <version>0.0.3</version>
</dependency>
For latest versions one could use: http://search.maven.org/#search%7Cga%7C1%7Cgcloud-java
If you want to run your code (not as a test via "mvn test") you have various options. here are some references:
http://www.vineetmanohar.com/2009/11/3-ways-to-run-java-main-from-maven/
http://mojo.codehaus.org/exec-maven-plugin/examples/example-exec-for-java-programs.html

There are other options such as using ant and ivy (http://ant.apache.org/ivy/history/2.2.0/tutorial/start.html) or using tools or various maven plugins to bundle all jars (latter is not recommended).

**Comment by aozarov** [comment]
@jgeewax let me know what you want to do about this one?

**Comment by jgeewax** [comment]
Is there any value is having a "starter-project" template (*not* the expense report example -- but one that is just a blank template) ?

**Comment by aozarov** [comment]
I think little value. All 3 main IDEs have an option for creating a "starter" maven project and an easy way to add dependencies. Yes, we could create a simple "hello gcloud java" simple project with maven, ant and gradle config files (already configured to pull gcloud-java) (we could also provide project configuration for all 3 main IDEs but I am not sure that is a great idea, especially when configuration may differ between IDEs versions) but I think one could also clone the "expense report" project (when available) and trim the unnecessary files. In any case, I don't object the idea (but not sure where exactly to put it - see: https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html -- as I don't think we need to create another gitHub project for it).

**Comment by jgeewax** [comment]
My worry is that someone starting like I did with the guide you pointed me to will hit the same ridiculous bug that I hit... A starter project is one way of solving that problem. Do you have ideas for others?

**Comment by aozarov** [comment]
Right, issue #36 could have been avoided by using a pom.xml template that specify Java source/target versions (but this is not great as one, for example, may want to use java 1.8 or later when they come).
Also, other maven related issues could appear when using such an old maven version.
However I just realized we could use maven-enforcer-plugin (http://maven.apache.org/enforcer/maven-enforcer-plugin/usage.html) in the pom template to specify minimum Java and Maven requirements (which will probably solve most of the configuration issues) so maybe not a bad idea after all (alternative would be just document it). Leaving this open so we can take action on it but unassigned it, to allow other to pick it up until I get to it.

**Comment by jgeewax** [comment]
Great so the deliverable here is... to somehow use the maven-enforcer plugin to require specific versions?

**Comment by aozarov** [comment]
Yes, though we could do that only in (1) our own pom file for anyone accessing our code via its source/gitHub (2) in the template/simple project that would have its own pom enforcing it.

Probably best would have been if we could someone provide such enforcement on the maven artifact (so anyone that adds us as a dependency would need to confirm to it) but I am not aware of a way to do that (or if that is even possible).

**Comment by aozarov** [comment]
After fix #78  it should be obvious if minimum requirements are not met
